File: app/src/main/java/com/uvnode/typhoon/sources/HttpSource.java
package com.uvnode.typhoon.sources;

import com.uvnode.typhoon.sources.executor.JSEClient;

import okhttp3.OkHttpClient;

public abstract class HttpSource extends MetaSource{

    private OkHttpClient okClient;
    private JSEClient jseClient;

    public OkHttpClient getOkClient() {
        return okClient;
    }

    public void setOkClient(OkHttpClient okClient) {
        this.okClient = okClient;
    }

    public void setJSEClient(JSEClient jseClient) { this.jseClient = jseClient; }
}

File: app/src/main/java/com/uvnode/typhoon/extensions/source/HttpSource.kt
package com.uvnode.typhoon.extensions.source

import com.uvnode.typhoon.extensions.executor.JSEClient
import okhttp3.OkHttpClient

abstract class HttpSource : MetaSource(), Rankable, Filterable {
    open var okClient: OkHttpClient? = null
    open var jseClient: JSEClient? = null
}